src/app/child-dev-project/notes/notes-related-to-entity/notes-related-to-entity.component.spec.ts

```diff
@@ -79,11 +81,52 @@ describe("NotesRelatedToEntityComponent", () => {
     component.entity = entity;
     component.ngOnInit();
     note = component.generateNewRecordFactory()();
-    expect(note.relatedEntities).toEqual([entity.getId(true)]);
+    expect(note.relatedEntities).toContain(entity.getId(true));
     expect(note.children).toEqual(["someChild"]);
     expect(note.schools).toEqual(["someSchool"]);
   });
 
+  it("should create a new note and fill it with indirectly related references (2-hop) of the types allowed for note.relatedEntities", () => {
+    @DatabaseEntity("TestNewNoteRelatedRefsEntity")
+    class TestNewNoteRelatedRefsEntity extends Entity {
```

Comment:
```suggestion
    class EntityWithRelations extends Entity {
```
I prefer to have names that explain what the purpose of the class is as easy as possible instead of having this schematic names that contain a lot of prefixes that are very hard to read.